packages/react/primitive/src/Primitive.tsx

```diff
@@ -46,9 +46,9 @@ const Primitive = NODES.reduce((primitive, node) => {
     const { asChild, ...primitiveProps } = props;
     const Comp: any = asChild ? Slot : node;
 
-    React.useEffect(() => {
+    if (typeof window !== 'undefined') {
       (window as any)[Symbol.for('radix-ui')] = true;
-    }, []);
+    }
```

Comment:
Note this change, this is to allow simple primitives to be server components (like `AccessibleIcon`)

No context for how it's used in the original PR, but seems benign to do the assignment in render:
https://github.com/radix-ui/primitives/pull/853